### PR TITLE
Add release notes for 2.4.0

### PR DESCRIPTION
**Pull request by madkinsz**
[Rendered](https://github.com/PrefectHQ/prefect/blob/release-2.4.0/RELEASE-NOTES.md#release-240)

### CODE REVIEW

File: RELEASE-NOTES.md [hunk 1]
```diff
@@ -1,5 +1,40 @@
 # Prefect Release Notes
 
+## Release 2.4.0
+
+### Exciting New Features 🎉
+- Add `ECSTask` infrastructure block to run commands and flows on AWS ECS<br />
+    See [the documentation](https://prefecthq.github.io/prefect-aws/ecs/) in the [prefect-aws collection](https://prefecthq.github.io/prefect-aws/) and usage notes in the [our infrastructure guide](https://docs.prefect.io/concepts/infrastructure/#ecstask)
```
Comment (tpdorsey) [inline RELEASE-NOTES.md:7]:
```suggestion
    See [the documentation](https://prefecthq.github.io/prefect-aws/ecs/) in the [prefect-aws collection](https://prefecthq.github.io/prefect-aws/) and usage notes in the [infrastructure guide](https://docs.prefect.io/concepts/infrastructure/#ecstask)
```

File: RELEASE-NOTES.md [hunk 2]
```diff
@@ -1,5 +1,40 @@
...
+
+### Enhancements
+- Update the deployments CLI to better support CICD use cases — https://github.com/PrefectHQ/prefect/pull/6697
```
Comment (tpdorsey) [inline RELEASE-NOTES.md:10]:
```suggestion
- Update the deployments CLI to better support CI/CD use cases — https://github.com/PrefectHQ/prefect/pull/6697
```

File: RELEASE-NOTES.md [hunk 3]
```diff
@@ -1,5 +1,40 @@
...
+- Protect against race condition with deployment schedules — https://github.com/PrefectHQ/prefect/pull/6673
+- Fix saving of block fields with aliases — https://github.com/PrefectHQ/prefect/pull/6758
+- Preserve task dependencies to futures passed as parameters in `.map` — https://github.com/PrefectHQ/prefect/pull/6701
+- Update task run orchestration to include latest metadata in context — https://github.com/PrefectHQ/prefect/pull/6791
+
+### Documentation
+- Task runner documentation fixes and clarifications — https://github.com/PrefectHQ/prefect/pull/6733
+- Add notes for Windows and Linux installation — https://github.com/PrefectHQ/prefect/pull/6750
+- Add a catalog of implementation recipes — https://github.com/PrefectHQ/prefect/pull/6408
+- Improve to storage and file systems documentation — https://github.com/PrefectHQ/prefect/pull/6756
```
Comment (tpdorsey) [inline RELEASE-NOTES.md:31]:
```suggestion
- Improve storage and file systems documentation — https://github.com/PrefectHQ/prefect/pull/6756
```

**Review by tpdorsey — APPROVED** [review]

**Review by anna-geller — APPROVED** [review]
IMO it would be great to note that the changes made in #6674 enabled new infrastructure blocks from Prefect Collections (as a mechanism), but LGTM as is

File: RELEASE-NOTES.md [hunk 4]
```diff
@@ -1,5 +1,40 @@
...
+
+### Enhancements
+- Update the deployments CLI to better support CI/CD use cases — https://github.com/PrefectHQ/prefect/pull/6697
+- Improve database query performance by removing unnecessary SQL transactions — https://github.com/PrefectHQ/prefect/pull/6714
+- Update blocks to dispatch instance creation using slugs — https://github.com/PrefectHQ/prefect/pull/6622
+- Add flow run start times to flow run metadata in UI — https://github.com/PrefectHQ/prefect/pull/6743
+- Update default infrastructure command to be set at runtime — https://github.com/PrefectHQ/prefect/pull/6610
+- Allow environment variables to be "unset" in infrastructure blocks — https://github.com/PrefectHQ/prefect/pull/6650
+- Add favicon switching feature for flow and task run pages — https://github.com/PrefectHQ/prefect/pull/6794
+- Update `Deployment.infrastructure` to accept types outside of the core library — https://github.com/PrefectHQ/prefect/pull/6674
```
Comment (madkinsz) [inline RELEASE-NOTES.md:17]:
```suggestion
- Update `Deployment.infrastructure` to accept types outside of the core library i.e. custom infrastructure or from collections — https://github.com/PrefectHQ/prefect/pull/6674
```
@anna-geller

**Pull request merged**